src/parsers/manifest/dash/node_parsers/SegmentTimeline.ts

```diff
@@ -18,24 +18,26 @@ import parseS, {
   IParsedS,
 } from "./S";
 
-export type IParsedTimeline = IParsedS[];
+export type ITimelineParser = () => IParsedS[];
 
 /**
  * @param {Element} root
- * @returns {Array.<Object>}
+ * @returns {Function}
  */
-export default function parseSegmentTimeline(root: Element) : IParsedTimeline {
-  const parsedS : IParsedS[] = [];
-  const timelineChildren = root.childNodes;
-  for (let i = 0; i < timelineChildren.length; i++) {
-    if (timelineChildren[i].nodeType === Node.ELEMENT_NODE) {
-      const currentElement = timelineChildren[i] as Element;
-
-      if (currentElement.nodeName === "S") {
+export default function parseSegmentTimeline(root: Element) : ITimelineParser {
+  let result : IParsedS[] | null = null;
+  return function() {
+    if (result === null) {
+      const parsedS : IParsedS[] = [];
+      const timelineChildren = root.getElementsByTagName("S");
```

Comment:
Is using `getElementsByTagName` really faster ?

Reply:
I don't know :p